app/objects/c_report.py

```diff
@@ -60,3 +60,142 @@ def store(self, ram):
         existing.update('hosts', self.hosts)
         existing.update('network_map', self.network_map)
         return existing
+
+    def retrieve_host_by_id(self, id):
+        return self.hosts[id]
+
+    # 
+    # Getters / Setters
+    # 
+
+    def get_host_freebie_abilities(self, id):
```

Comment:
If you're looking for a slightly more Pythonic way to do these, I recommend looking into Python properties: https://www.scaler.com/topics/python/python-property/

Offers a way to implement getters and setters that gives you a bit more freedom with how to use them. Might not be as applicable in this case given how many parameters you're using for these. Just something to think about.